docs/en/resource/supplemental.md

```diff
@@ -0,0 +1,351 @@
+
+```ts
+const memoryTemplate: DataTemplate = {
+	read: ({ query }, get, put) => {
+		// get the data from the resource
+		let data: any[] = get();
+
+		// filter the data
+		const filteredData = filter && query ? data.filter((i) => filter(query, i)) : data;
+		// put the filtered data into the resource so it
+		// won't try and read it again for the same query
+		put(0, filteredData);
+		// return the data that was requested
+		return { data: filteredData, total: filteredData.length };
+	}
+};
+
+export default factory(function() {
+	// pass a resource factory and the data to sideload
+	// instead of just the resource
+	return (
+		<List
+			resource={{
+				resource: () => createResource(memoryTemplate),
+				data: ['dog', 'fish', 'cat']
+			}}
+		/>
+	);
+});
+```
+
+The above approach will allow the `data` to be changed at any time causing the resource to be cleared and the new data sideloaded again. This is the approach that should be used in most in memory data array scenarios and can be seen being used within our `TimePicker` widget when time options are generated and passed to the `Select`.
+
+# Data Middleware
+
+The data middleware provides a widget with access to the underlying resource via the data middleware which offers an api to initialise and share the resource and functions to interact with it. The middleware should be imported and passed to the widgets `create` function the same as other middlwares. It will add two extra properties to your widget; `resource` and `transform`.
+
+```ts
+import { create } from '@dojo/framework/core/vdom';
+import { createDataMiddleware } from '@dojo/framework/core/middleware/data';
+
+const factory = create({ data });
+
+export const DataAwareWidget = factory(function ({
+	middleware: { data }
+}) {
+	const api = data();
+}
+```
+
+## The Resource Property
+
+The resource property added to a widget can be used to pass a resource, a resource factory and data to initialise the resource with or a resource wrapper.
+
+### resource
+
+A resource is the result of the `createResource` function. It is a new resource and does not have any `ResourceOptions` or data middleware connections associated to it. This is how you will pass a resource into a widget in most cases pertaining to a remote data source.
+
+```tsx
+import { DataTemplate, createResource } from '@dojo/framework/core/resource';
+import { fetcher } from './personfetcher';
+import { List } from './List';
+
+const template: DataTemplate = {
+	read: fetcher
+};
+
+const resource = createResource(template);
+
+export default factory(function() {
+	return <List resource={resource} />;
+});
+```
+
+### resource factory with data
+
+A resource factory with data is the approach you would use when you wish to side load data into the resource. This will commonly be used when working in an in memory data array and a memory template as it allows the resource data to be reloaded when the data passed is changed.
```

Comment:
```suggestion
A resource factory with data is the approach used when data needs to be sideloaded into the resource. This will commonly be used when working with an in-memory data array and a memory template, as it allows the resource data to be reloaded when the provided data is changed.
```

Reply:
not sure if the first sentence should end 'sideloaded into the resource' or 'sideloaded into the widget'.

Reply:
resource, you were correct